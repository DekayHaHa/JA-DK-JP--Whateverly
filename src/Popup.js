import React, { Component } from 'react';

export default class Popup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showGenres: false,
      genreData: []
    };
  }

  closePopup = () => {
    this.props.setPopup( false );
  }

  getGenreDescript = () => {
    let allGenres = this.props.mainGenres.filter(ea => {
      return this.props.currentGenres.includes(ea.genre);
    });
    this.setState({genreData: allGenres, showGenre: true})
  }

  render() {
    const styleImg = { backgroundImage: 'url(' + this.props.currentImage + ')', backgroundSize: 'cover',  };
    if(this.state.showGenre) {
      return (
        <div className="popup-overlay">
          <div id="genre" className="popup">
            <button className="close" onClick={this.closePopup}>X</button>
              {this.state.genreData.map(genre => {
                return (
                  <div>
                    <h3>{genre.genre}</h3>
                    <p>{genre.description}</p>
                  </div>) 
                })}
              <button className="showGame" onClick={() => {this.setState({showGenre: false})}}>Show Game</button>
          </div>
        </div>
      )
    }
    return (
      <div className="popup-overlay">
        <div className="popup">
        <button className="close" onClick={this.closePopup}>X</button>
          <div className="popup-img-container">
            <section className="popup-img" style={styleImg}></section>
          </div>
          <article className="popup-text-container">
            <h3 className="popup-title">Game Title: 
              <span className="value"> {this.props.currentTitle}</span></h3>
            <h4 className="popup-score">MetaCritic Score: 
              <span className="value"> {this.props.currentScore}</span></h4>
            <h4 className="popup-multi">Multiplayer: 
              <span className="value"> {(this.props.currentMulti).toString()}</span></h4>
            <h4 className="popup-plat">Available On: 
              <span className="value"> {this.props.currentPlatforms.join(" - ")}</span></h4>
            <h4 className="popup-genre">Genres: 
              <span className="value genre"> {this.props.currentGenres.join(" - ")}</span></h4>
              <button className="showGenre" onClick={this.getGenreDescript}>Show Genre Details</button>
          </article>
        </div>
      </div>
    )
  }
}