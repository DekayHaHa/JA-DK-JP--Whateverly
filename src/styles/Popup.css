.popup-overlay {
  background-color: rgba(0, 100, 100, .7);
  position: absolute;
  z-index: 5;
  width: 100%;
  height: 100%;
  margin-top: -30px;
}

.popup {
  background-color: gray;
  box-shadow: 5px 5px 5px black;
  position: absolute;
  height: 60%;
  width: 60%;
  margin: 7% 20%;
  border-radius: 20px;
}