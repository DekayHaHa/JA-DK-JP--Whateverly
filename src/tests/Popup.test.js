import React from 'react';
import ReactDOM from 'react-dom';
import Popup from '../Popup';
import { shallow } from 'enzyme';

describe('Popup', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Popup />
    );
  });
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
})