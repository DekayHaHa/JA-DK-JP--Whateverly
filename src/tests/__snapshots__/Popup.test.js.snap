// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should match snapshop 1`] = `
<div
  className="popup-overlay"
>
  <div
    className="popup"
  >
    <button
      className="close"
      onClick={[Function]}
    >
      X
    </button>
    <div
      className="popup-img-container"
    >
      <section
        className="popup-img"
        style={
          Object {
            "backgroundImage": "url(https://i.ebayimg.com/images/g/ZX4AAOSw1vlUrlTm/s-l300.jpg)",
            "backgroundSize": "cover",
          }
        }
      />
    </div>
    <article
      className="popup-text-container"
    >
      <h3
        className="popup-title"
      >
        Game Title:
        <span
          className="value"
        >
           
          Skyrim
        </span>
      </h3>
      <h4
        className="popup-score"
      >
        MetaCritic Score:
        <span
          className="value"
        >
           
          99
        </span>
      </h4>
      <h4
        className="popup-multi"
      >
        Multiplayer:
        <span
          className="value"
        >
           
          false
        </span>
      </h4>
      <h4
        className="popup-plat"
      >
        Available On:
        <span
          className="value"
        >
           
          PC - Switch
        </span>
      </h4>
      <h4
        className="popup-genre"
      >
        Genres:
        <span
          className="value genre"
        >
           
          Adventure - Action
        </span>
      </h4>
      <button
        className="showGenre"
        onClick={[Function]}
      >
        Show Genre Details
      </button>
    </article>
  </div>
</div>
`;
